### PR TITLE
Refactor/controllers

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?

https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/rails_refactor
Because the controller was verbose and had lots of repetition
Because this project needed some tests
### What Changed?

Refactored ItemsController
Moved logic from Controller to Model
Added rspec for new Model methods

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -1,10 +1,19 @@
 class Item < ActiveRecord::Base
   belongs_to :category
 
+  validates :name,            presence: true
```
Comment (albaer) [inline app/models/item.rb:4]: Nice. These validations definitely belong in the model.

File: Gemfile [hunk 2]
```diff
@@ -20,3 +20,10 @@ group :development, :test do
   gem 'spring'
 end
 
+group :test do
```
Comment (albaer) [inline Gemfile:23]: Tests! Yay!

File: app/controllers/items_controller.rb [hunk 3]
```diff
@@ -1,36 +1,13 @@
 class ItemsController < ApplicationController
   def index
-    @category_list = Category.all.map{ |category| [category.name, category.id] }
-    @valid_items = Item.all.select{ |item| item.expires > Date.today }
+    @category_list = Category.all_names
```
Comment (albaer) [inline app/controllers/items_controller.rb:3]: Where is this used?

File: app/models/category.rb [hunk 4]
```diff
@@ -1,3 +1,7 @@
 class Category < ActiveRecord::Base
   has_many :items
+
+  def self.all_names
```
Comment (albaer) [inline app/models/category.rb:4]: This looks like something that is going to get used for a select menu somewhere... is that right? If so, there are some handy rails things we can use for that.

**Comment by albaer** [comment]
:100: on the tests!
:shipit: and we can take a look at some methods that will help with the select stuff.

**Comment by albaer** [comment]
Also, "Moved logic from Controller to Model" is one of my favorite things to see in a pr comment :)

File: app/controllers/items_controller.rb (hunk 3 above)
Comment (TylerRockwell) [inline app/controllers/items_controller.rb:3]: It is used to populate the category select box on the item form.

File: app/models/category.rb (hunk 4 above)
Comment (TylerRockwell) [inline app/models/category.rb:4]: Yes. I'm all for handy rails things! Please, teach me your ways!

**Pull request merged**